### DISCUSSION TITLE
Format of the timestamp in timestamp extrinsic?

### DISCUSSION THREAD

**Discussion by stolpa4** [polkascan/py-substrate-interface#301] [Q&A] [answered]
Hello, I found that for recent blocks (polkadot) the timestamp extrinsic is parsed with `extrinsic['call']['call_args']['type'] == 'Moment'` and value is some rather big number (e.g. for block 7313001 it is 1634560404001) - clearly it is not a Unix timestamp. How can I parse this number into timestamp? 

I've used to parse timestamp values as strings in format `'%Y-%m-%dT%H:%M:%S.%f'`, but it worker for timestamps with `extrinsic['call']['call_args']['type'] == 'Compact<Moment>'`.

Now when the call value is integer, I need to introduce some changes to my code, but I can't find what format the timestamp value is.

**Comment by maayank** [comment, marked as answer]
It’s a unixtimestamp in milliseconds

**Comment by maayank** [comment]
If you don’t care for ms precision then simply divide by 1000 and round down

**Comment by stolpa4** [comment]
@maayank Thanks!

**Comment by arjanz** [comment]
@stolpa4 yes this is a inconsistency we have to fix, so the result with all metadata  versions will be the same. I will reopen this issue to convert a `Compact<Moment>` type in Metadata V14 to a Python datetime and serialize it to a string format like before.

**Comment by arjanz** [comment]
After some discussion how to deal with the `Moment` type it will be consistently treated as an `u64` as per the Substrate codebase, also for pre-MetadataV14 runtimes. Conversion to e.g. a datetime will be up to the application.
